backends/graphics/opengl/opengl-graphics.cpp

```diff
@@ -1212,41 +1214,13 @@ bool OpenGLGraphicsManager::saveScreenshot(const Common::String &filename) const
 	pixels.resize(lineSize * height);
 	GL_CALL(glReadPixels(0, 0, width, height, GL_RGB, GL_UNSIGNED_BYTE, &pixels.front()));
 
-#ifdef USE_PNG
 	const Graphics::PixelFormat format(3, 8, 8, 8, 0, 16, 8, 0, 0);
```

Comment:
Are you sure this is endian-safe? It seems to me `glReadPixels(.., GL_RGB, GL_UNSIGNED_BYTE, ..)` gives us pixels in RGB byte order. But `Graphics::PixelFormat(3, 8, 8, 8, 0, 16, 8, 0, 0);` is RGB in word order. I would expect something like:
```cpp
#ifdef SCUMM_LITTLE_ENDIAN
const Graphics::PixelFormat format(3, 8, 8, 8, 0, 16, 8, 0, 0);
#else
const Graphics::PixelFormat format(3, 8, 8, 8, 0, 0, 8, 16, 0);
#endif
```

Reply:
Does that mean that saving PNG screenshots is also broken on big endian?

Reply:
I think PNG worked on BE by accident. See:
https://github.com/scummvm/scummvm/blob/22f12e5ffb91583a0955d0e8a637aff1089459d2/image/png.cpp#L266

`PNG_COLOR_TYPE_RGB` means the PNG encoder expects byte order RGB pixels. Which happens to be exactly what OpenGL provides. The PixelFormat is incorrect but the actual pixel data is correct.